### DISCUSSION TITLE
Related posts not showing up despite having two posts with the same tag

### DISCUSSION THREAD

**Discussion by jjets718** [GetPublii/Publii#1521] [Help]
Hi there! I'm using **[this theme](https://marketplace.getpublii.com/themes/square/)** in Publii.

Right now, I have two posts that each have the tag "X".

I've attached what my related post settings look like under the "Site settings" tab.

With this theme, 2 posts using the tag "X", and the attached related post settings, each post is not displaying the other related post at the bottom. Did I mess something up or is this a theme limitation?

I would be so grateful for any help. Thanks!

<img width="1440" alt="Screenshot 2024-05-24 at 9 23 47 PM" src="https://github.com/GetPublii/Publii/assets/994615/479681ad-59e5-4bdf-8fa3-84070bc17134">

**Comment by 3muchfun** [comment]
From your screenshot I would suggest to activate the "Related post options" by clicking on the switch.

**Reply by jjets718** [reply]
Thanks! I activated the "Related post options" setting, pushed the change to my hosting, and I still don't have any "Related posts" on any of the three posts (I published one additional post).

**Reply by 3muchfun** [reply]
Hi, @jjets718 , it seems that I have the same issue when using that Square 2.5.0.0 theme. In the App the switch is set to show related posts, but in the config.json the renderRelatedPosts is set to false...Please refer to the screenshot.
@bobmitro  or @dziudek should have look on that, I assume.

<img width="1215" alt="Bildschirmfoto 2024-06-02 um 17 21 52" src="https://github.com/GetPublii/Publii/assets/17292904/f2474e98-ee4e-4f06-8a2a-157f4f74062a">

**Reply by jjets718** [reply]
@3muchfun Ah thank you so much! I'm glad the issue isn't with something I did. I've been playing around with the theme code to try and get related posts to work but to no avail so far. If I figure it out, I'll follow up here.

**Comment by bobmitro** [comment]
Hey, 

Thanks to @jjets718 engagement and donation, the **Square theme** in version 2.6.0.0 now includes the "Related posts" option.
🔗 [Check it out here](https://marketplace.getpublii.com/themes/square/)

**Comment by nordseo** [comment]
I'm also having an issue with related posts not appearing in Mono Theme. [demo looks like it works](https://demo.getpublii.eu/themes/mono/intricacies-of-geometric-design/)

```
"renderer": {
        "renderRelatedPosts": true,
        "relatedPostsNumber": 10000,
```

I've experimented with using the same MainTag, and shared tags; I also have the theme options Related Posts set to active, but they are still not showing in the related posts section. 

(I don't have featured images, authors, comments, post dates activated.)

**Reply by nordseo** [reply]
The documentation and examples of Publii's functions are minimal.

Posts that share the same tag, and main tag, simply do not appear in Related Posts.

**Reply by nordseo** [reply]
https://forum.getpublii.com/topic/related-posts-in-simple/#:~:text=The%20related%20posts%20are%20created,related%20posts%20are%20not%20created.

This doesn't really explain much about how related posts work.

**Reply by nordseo** [reply]
Can't seem to find anything regarding relatedPostCriteria options or customizations:

https://www.google.com/search?q=Publii+%22relatedPostsCriteria%22
https://github.com/GetPublii/Publii/discussions?discussions_q=relatedPostsCriteria

Here, there are no examples of how to use the function efficiently, such as getting posts by the current tag, main tag, or by post-template.

https://getpublii.com/dev/getpostsbytags-helper/

**Reply by nordseo** [reply]
This would be great for displaying related posts from specific tags, but the problem is that it doesn't support posts. Many of Publii's added features should have been for posts, not the theme core. Like repeated fields, post-dropdown, tags-dropdown. However, none of these support postConfig, which seems like Publii is missing 90% of its potential. 

https://getpublii.com/dev/tags-dropdown/

**Reply by nordseo** [reply]
![Screenshot 2024-09-03 at 2 24 09 PM](https://github.com/user-attachments/assets/e3418899-098f-476b-b97f-c38699be6525)

As an example of the lack of flexibility by only supporting post-dropdowns and tags-dropdowns for the theme Option but not for postConfig. We have a series of hidden CTA (call to action) posts featured on specific pages. 

The missing feature is that we must create each item manually by code first, leaving limited slots. Only after they are registered in the theme options can they be selected with a dropdown in the post. But this is highly cumbersome and time-consuming. 

When I edit a post, I should be able to add a post-dropdown field and a tags-dropdown field.

It just doesn't make sense to only have these fields support the theme when 90% of their use would be for postConfig.

**Reply by jjets718** [reply]
@nordseo Thanks for sharing your experience. I have no affiliation with Publii but I like the platform and want it to succeed.

I can't speak to the related posts issue for the Mono Theme.

However, if you have feature ideas or functionality you'd like them to build, I recommend reaching out to them.

If you want them to build a feature that they'll make available to all Publii users, in my experience, they'll give you an estimate for how many hours of work that it will take and ask you to donate to Publii in the amount of that many hours multiplied by their hourly rate. I thought this seemed like a fair exchange.

If you want them to build out features only for you and/or your clients, you can hire them here: https://getpublii.com/customization-service/

Their hourly rate is very reasonable.

**Reply by nordseo** [reply]
Thanks. I do believe that more documentation of examples would be very useful. I was able to build a lot of customized solutions.
